service/controller/resource/monitoring/prometheus/resource.go

```diff
@@ -431,5 +383,5 @@ func hasChanged(current, desired metav1.Object) bool {
 	c := current.(*promv1.Prometheus)
 	d := desired.(*promv1.Prometheus)
 
-	return !reflect.DeepEqual(c.Spec, d.Spec)
+	return !cmp.Equal(c.Spec, d.Spec, cmpopts.EquateEmpty(), cmpopts.IgnoreFields(promv1.PrometheusSpec{}, "RemoteWrite"))
```

Comment:
Why do we need to add `cmpopts.EquateEmpty()` here ?

Reply:
for computing diff of pointers fields, which means empty fields will be equal.

Reply:
``` 
// EquateEmpty returns a Comparer option that determines all maps and slices
// with a length of zero to be equal, regardless of whether they are nil.
``` 

for me it's fine, we can remove it. I don't see a strong case!

Reply:
Ok, let's please remove this then.